File: src/main/java/com/moira/itda/domain/common/controller/CommonCodeController.java
package com.moira.itda.domain.common.controller;

import com.moira.itda.domain.common.dto.request.CommonCodeAddRequest;
import com.moira.itda.domain.common.dto.response.CommonCodeDetailResponse;
import com.moira.itda.domain.common.dto.response.CommonCodeResponse;
import com.moira.itda.domain.common.service.CommonCodeService;
import com.moira.itda.global.aop.IsAdmin;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.net.URI;
import java.util.List;

@RequiredArgsConstructor
@RestController
public class CommonCodeController {
    private final CommonCodeService commonCodeService;

    /**
     * 공통코드 > 저장
     */
    @IsAdmin
    @PostMapping("/api/common/code")
    public ResponseEntity<Object> add(@RequestBody CommonCodeAddRequest request) {
        commonCodeService.add(request);

        return ResponseEntity.created(URI.create("/api/common/code")).build();
    }

    /**
     * 공통코드 > 단건 조회
     */
    @GetMapping("/api/common/code/{key}")
    public ResponseEntity<List<CommonCodeDetailResponse>> get(@PathVariable String key) {
        List<CommonCodeDetailResponse> list = commonCodeService.get(key);

        return ResponseEntity.ok(list);
    }

    /**
     * 공통코드 > 전체 조회
     */
    @IsAdmin
    @GetMapping("/api/common/code")
    public ResponseEntity<List<CommonCodeResponse>> getAll() {
        List<CommonCodeResponse> list = commonCodeService.getAll();

        return ResponseEntity.ok(list);
    }

    /**
     * 공통코드 > 삭제
     */
    @IsAdmin
    @DeleteMapping("/api/common/code/{key}")
    public ResponseEntity<Object> delete(@PathVariable String key) {
        commonCodeService.delete(key);

        return ResponseEntity.ok(null);
    }
}

File: src/main/kotlin/com/moira/itda/domain/common_code/controller/CommonCodeController.kt
package com.moira.itda.domain.common_code.controller

import com.moira.itda.domain.common_code.dto.response.AllCodeDetailResponse
import com.moira.itda.domain.common_code.dto.response.CodeDetailResponse
import com.moira.itda.domain.common_code.service.CommonCodeService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

/**
 * 공통코드 조회
 */
@RestController
class CommonCodeController(
    private val service: CommonCodeService
) {
    /**
     * 공통 > 공통코드 조회
     */
    @GetMapping("/api/code/{key}")
    fun getCode(@PathVariable key: String): ResponseEntity<List<CodeDetailResponse>> {
        val response = service.getCode(key = key)

        return ResponseEntity.ok(response)
    }

    /**
     * 공통 > 공통코드 일괄 조회
     */
    @GetMapping("/api/code/all")
    fun getAllCode(
        @RequestParam(required = true) code1: String,
        @RequestParam(required = false) code2: String?,
        @RequestParam(required = false) code3: String?,
        @RequestParam(required = false) code4: String?,
        @RequestParam(required = false) code5: String?,
    ): ResponseEntity<AllCodeDetailResponse> {
        val response = service.getAllCode(keyList = listOf(code1, code2, code3, code4, code5))

        return ResponseEntity.ok(response)
    }
}